openadapt/ml_models/completion_factory.py

```diff
@@ -0,0 +1,7 @@
+
+from base_provider import Modality, CompletionProvider
+class CompletionProviderFactory:
+    def get_for_modality(modality: Modality) -> list[CompletionProvider]:
```

Comment:
What do you think about moving this to CompletionProvider and making it a `classmethod` (or staticmethod) as well?